renderdoc/driver/gl/gl_driver.h

```diff
@@ -279,6 +279,11 @@ class WrappedOpenGL : public IFrameCapturer
     // bound to the second.
     GLuint renderbufferReadTex;
     GLuint renderbufferFBOs[2];
+
+    // since compressed textures cannot be read back on GLES we have to store them during the
+    // uploading
+    // target -> level -> data
+    map<int, map<int, vector<byte> > > compressedData;
```

Comment:
Since you're already handling uploading 2D arrays one slice at a time, could you handle it that way? ie. convert the individual cubemap face enum to an array index from 0 to 5 and modify zoffset internally, then treat everything the same.